_build_backend/vllm.py

```diff
@@ -0,0 +1,195 @@
+envs = load_module_from_path('envs', os.path.join(ROOT_DIR, 'vllm', 'envs.py'))
+VLLM_TARGET_DEVICE = envs.VLLM_TARGET_DEVICE
+VLLM_USE_PRECOMPILED = envs.VLLM_USE_PRECOMPILED
+
+
+def _get_requires_for_build_extensions() -> list[str]:
+    """ returns the requirements for extensions builds"""
+    with open("requirements-build.txt") as fh:
+        return [line.strip() for line in fh.readlines()]
+
+
+def _get_requires_for_basic_build() -> list[str]:
+    """ returns the base requirements"""
+    return [
+        req for req in _get_requires_for_build_extensions()
+        if "setuptools" in req
+    ]
+
+
+def _check_for_env_var(key: str, expected_value: str, strict: bool = False):
+    """Print a warning when the env var's value doesn't match the expected value.
+
+    When strict is set to True, raises SetupError instead of warning.
+    """
+    value = os.getenv(key)
+    if value and value == expected_value:
+        return
+
+    warning = (
+        f"{key} is not defined, but {'is' if strict else 'might be'} required for this build."
+        if value is None else
+        f"{key} is set to {value}, but {expected_value} is suggested.")
+
+    if strict:
+        from setuptools.errors import SetupError
+
+        raise SetupError(warning)
+
+    msg = dedent(
+        """
+        ***
+        {warning}
+        If the build fails, try setting
+
+            {key}={suggested_value}
+
+        in your environment before starting the build.
+        ***""", )
+
+    import warnings
+
+    warnings.warn(
+        msg.format(warning=warning, key=key, suggested_value=expected_value),
+        stacklevel=2,
+    )
+
+
+def _check_for_extra_index_url(expected_value: str, strict: bool = False):
+    """Print a warning when the env var's value doesn't match the expected value.
+
+    When strict is set to True, raises SetupError instead of warning.
+    """
+    has_uv = which("uv")
+    if has_uv:
+        _check_for_env_var("UV_EXTRA_INDEX_URL", expected_value, strict=strict)
+        # need to match pip's index behaviour,
+        # see https://docs.astral.sh/uv/pip/compatibility/#packages-that-exist-on-multiple-indexes
+        _check_for_env_var("UV_INDEX_STRATEGY",
+                           "unsafe-best-match",
+                           strict=strict)
+    else:
+        _check_for_env_var("PIP_EXTRA_INDEX_URL",
+                           expected_value,
+                           strict=strict)
+
+
+def get_requires_for_build_wheel(config_settings=None) -> list[str]:
+    """ Dynamically computes the wheel build requirements based on VLLM_TARGET_DEVICE
+
+    torch versions here will have to be kept in sync with the corresponding `requirements-<device>.txt` files.`
+    """
+    requirements_extras: list[str] = []
+
+    if VLLM_TARGET_DEVICE == "cpu" or VLLM_TARGET_DEVICE == "openvino":
+        _check_for_extra_index_url("https://download.pytorch.org/whl/cpu")
```

Comment:
i think it'd be good to extract the URL's into a dictionary with the keys being the name of the backend.